.github/workflows/docker-buld.yml

```diff
@@ -0,0 +1,38 @@
+name: Docker CI workflow 
+
+on:
+  push:
+    branches: ["main"]
+  pull_request:
+    branches: ["main"]
+
+jobs:
+
+  docker-build-prod:
+    runs-on: ubuntu-latest
+    name: Run docker build  
+    steps:
+      
+      - uses: actions/checkout@v4
+
+      - name: Docker Login
+        uses: docker/login-action@v3.0.0
+        with:
+          username: ${{ secrets.DOCKER_USER }}
+          password: ${{ secrets.DOCKER_TOKEN }}
+
+      - name: create docker image name
+        id: dockerimage
```

Comment:
the id should be unique and easier to understand
Suggestion: create_docker_image